### PR TITLE
Split CLI tests into basic usage and integration tests.

### PR DESCRIPTION
**Pull request by rabbah**
@fxulusoy FYI this is a result of ignoring a test the `http` test.
@mdeuser @dubeejw I split the basic cli tests into one that are integration tests and ones that are mostly basic cli usage - i expect we can put other tests in the new suite over time.

### CODE REVIEW

**Comment by rabbah** [comment]
@ChrBi @markusthoemmes a few changes for your review.

**Comment by rabbah** [comment]
@mdeuser @dubeejw note and review the changes in restructuring the basic cli tests - i moved some out of the system package and into a new cli test package intended to test basic operations and corner cases using the cli. not all of the tests in the latter can run without a deployed backend (yet).

File: tests/src/whisk/core/entity/test/SchemaTests.scala [hunk 1]
```diff
@@ -317,9 +321,9 @@ class SchemaTests extends FlatSpec with BeforeAndAfter with Matchers {
 
     it should "properly deserialize JSON" in {
         val json = Seq[JsValue](
-            JsObject("timeout" -> TimeLimit.STD_DURATION.toMillis.toInt.toJson, "memory" -> MemoryLimit.STD_MEMORY.toJson, "logs" -> LogLimit.STD_LOGSIZE.toJson),
-            JsObject("timeout" -> TimeLimit.STD_DURATION.toMillis.toInt.toJson, "memory" -> MemoryLimit.STD_MEMORY.toJson, "logs" -> LogLimit.STD_LOGSIZE.toJson, "foo" -> "bar".toJson),
-            JsObject("timeout" -> TimeLimit.STD_DURATION.toMillis.toInt.toJson, "memory" -> MemoryLimit.STD_MEMORY.toJson))
+            JsObject("timeout" -> TimeLimit.STD_DURATION.toMillis.toInt.toJson, "memory" -> MemoryLimit.STD_MEMORY.size.toInt.toJson, "logs" -> LogLimit.STD_LOGSIZE.size.toInt.toJson),
```
Comment (markusthoemmes) [inline tests/src/whisk/core/entity/test/SchemaTests.scala]: Use `toMB` instead of `size` to force this to return the value in MB (as expected by the backend)

File: tests/src/whisk/core/entity/test/SchemaTests.scala [hunk 2]
```diff
@@ -335,13 +339,13 @@ class SchemaTests extends FlatSpec with BeforeAndAfter with Matchers {
             JsObject(),
             JsNull,
             JsObject("timeout" -> TimeLimit.STD_DURATION.toMillis.toInt.toJson),
-            JsObject("memory" -> MemoryLimit.STD_MEMORY.toJson),
-            JsObject("logs" -> (LogLimit.STD_LOGSIZE + 1).toJson),
-            JsObject("TIMEOUT" -> TimeLimit.STD_DURATION.toMillis.toInt.toJson, "MEMORY" -> MemoryLimit.STD_MEMORY.toJson),
-            JsObject("timeout" -> (TimeLimit.STD_DURATION.toMillis.toDouble + .01).toJson, "memory" -> (MemoryLimit.STD_MEMORY.toDouble + .01).toJson),
+            JsObject("memory" -> MemoryLimit.STD_MEMORY.size.toInt.toJson),
+            JsObject("logs" -> (LogLimit.STD_LOGSIZE.size.toInt + 1).toJson),
```
Comment (markusthoemmes) [inline tests/src/whisk/core/entity/test/SchemaTests.scala]: see above

File: tests/src/whisk/core/entity/test/SchemaTests.scala [hunk 3]
```diff
@@ -335,13 +339,13 @@ class SchemaTests extends FlatSpec with BeforeAndAfter with Matchers {
...
+            JsObject("TIMEOUT" -> TimeLimit.STD_DURATION.toMillis.toInt.toJson, "MEMORY" -> MemoryLimit.STD_MEMORY.size.toInt.toJson),
+            JsObject("timeout" -> (TimeLimit.STD_DURATION.toMillis.toDouble + .01).toJson, "memory" -> (MemoryLimit.STD_MEMORY.size.toDouble + .01).toJson),
             JsObject("timeout" -> null, "memory" -> null),
             JsObject("timeout" -> JsNull, "memory" -> JsNull),
-            JsObject("timeout" -> TimeLimit.STD_DURATION.toMillis.toString.toJson, "memory" -> MemoryLimit.STD_MEMORY.toString.toJson))
+            JsObject("timeout" -> TimeLimit.STD_DURATION.toMillis.toString.toJson, "memory" -> MemoryLimit.STD_MEMORY.size.toInt.toString.toJson))
```
Comment (markusthoemmes) [inline tests/src/whisk/core/entity/test/SchemaTests.scala]: See above (apply to all instances)

**Comment by markusthoemmes** [comment]
Great job, looks good to me modulo comments.

File: tests/src/whisk/core/entity/test/SchemaTests.scala (hunk 1 above)
Comment (rabbah) [inline tests/src/whisk/core/entity/test/SchemaTests.scala]: good point - i used size because i assumed/knew it was MB already but that may change.

**Comment by mdeuser** [comment]
@rabbah - Thanks for the heads up.  When adding new test cases in the future, what litmus test can be applied to determine whether the test case falls into the "integration" camp or not?

**Comment by rabbah** [comment]
new IPG 386 and PPG 939.

**Comment by rabbah** [comment]
@mdeuser I split the tests based on what criteria and invariant were being checked. Basic usage, testing parameter corner cases,  validity of arguments and the like I put in the new suite. Tests which are verifying the CRUD+activate life cycles or package operations I left in the `system` package. We can evolve a better criteria for where to locate CLI tests.

**Comment by rabbah** [comment]
Going to rebase and resolve conflicts following merge of #927.

**Comment by rabbah** [comment]
@mdeuser @dubeejw I rebased and resolved conflicts from the tests added in #927. In the process I eliminated one test which was not redundant (checking that unset of multiple values works as expected).

One test is still using python CLI (down from 2) because the go CLI assumes https always even if api host is set explicitly. Can you address this?

**Comment by rabbah** [comment]
@mdeuser @dubeejw I rebased and resolved conflicts from the tests added in #927. In the process I discovered that doing `wsk --apihost <valid host> property get --apibuild` is not working and neither is the test which tries to access controller via `http`. Can you address these separately (https://github.com/openwhisk/openwhisk/issues/586) or send a pull request to my branch and I can merge here.

I also eliminated one test that is redundant (checking that unset of multiple values works as expected).

**Comment by rabbah** [comment]
PPG 945.

**Comment by rabbah** [comment]
Rebased and squashed.

**Comment by mdeuser** [comment]
@rr - You are correct in that that doing `go-cli/wsk --apihost <valid host> property get --apibuild` will not behave like the Python API.

The hard Go CLI limitation is support for argument overloading - assigning the same named argument to the root command and also to a sub-command.  In this example, not only is the argument assigned twice, but it has different characteristics.  The `--apihost` argument assigned to the `wsk` root command takes a value while the `--apihost` argument assigned to the `property get` command does not.  The Go CLI argument package permits the assignment of the same argument name to different sub-commands as well as the root command; however, for any particular CLI invocation, all arguments are merged such that the argument is applied only to the last sub-command.

If we want the `wsk property get` command to take a apihost argument as the target host, the argument label will need to change.  That said, I think the better solution may be to separate the existing `wsk property` command into two commands - one for configuring the CLI and another for accessing whisk server values.  This would remove the `--apihost` argument overload conflict.

**Comment by rabbah** [comment]
Making it too commands makes it inconvenient to check build versions against different deployments (now it's two steps). What about if get didn't accept arguments with -- but instead a set of predefined values so instead of --apihost it becomes apihost?

**Comment by rabbah** [comment]
It looks like we will need to address game outstanding issue separately (for http, missing host, and property get). In which case shall we merge this to get the suite split and accept that three tests will continue to run against the Python cli until a resolution is in place for the go cli?

**Comment by mdeuser** [comment]
Rebased and merged.  PG 79 is now running against latest blue/open

**Comment by rabbah** [comment]
Merged as https://github.com/openwhisk/openwhisk/commit/461d9a8fd67113bafd09ff4cfd09cc67780b219a.
